### PR TITLE
Begin with blank xklb-metadata.db from template? [w/ key columns added for Calibre-Web videos]

### PR DESCRIPTION
**Pull request by deldesir**
Add a blank xklb-metadata.db with all key columns for when Calibre-Web is enhanced with videos.

~Smoke tested on Ubuntu 24.04~

@holta

### CODE REVIEW

**Comment by holta** [comment]
Thanks @deldesir:

Can you explain more, what is the purpose & need here?

Also, if indeed this is truly needed (for now?) follow-up question(s) are:

- How would xklb-metadata.db blank db snapshots be maintained over time, as xklb's schema also evolves over time?

- Can the creation of blank db xklb-metadata.db snapshots be fully automated in future — instead of using a static template/xklb-metadata.db ?

ASIDE: It's true that having static snapshots-in-time (a museum or archive of past schemas!) doesn't hurt — e.g. reference / sample versions of xklb-metadata.db as they evolve — to help people visualize & understand, e.g. using https://dbhub.io and https://datasette.io, how this schema has progressed over time.

**Comment by holta** [comment]
Just fyi regardless of the long-term requirements (i.e. architectural design choices!), this PR would appear to be helpful during the current testing of these 5 PR's:

- PR iiab/calibre-web#253
- PR iiab/calibre-web#254
- PR iiab/calibre-web#256
- PR iiab/calibre-web#257
- PR iiab/calibre-web#258

**Comment by deldesir** [comment]
> Can you explain more, what is the purpose & need here?
Some columns (like media.live_status or media.error) are created dynamically until needed by xklb tool. We want to avoid altering the schema ourselves. This PR ensures the key columns are already present and fixes the errors displayed in the below screenshot.
WhatsApp Image 2024-09-25 at 10 01 23_62cb083c
 
> How would xklb-metadata.db blank db snapshots be maintained over time, as xklb's schema also evolves over time?

Xklb's schema is evolving in a per feature basis, not as a whole. The video-based subset is the only in use in our case and we go in par with changes made to it.

> Can the creation of blank db xklb-metadata.db snapshots be fully automated in future — instead of using a static template/xklb-metadata.db ?

Yes, this is possible if a maintaining an up-to-dated static template is not practical enough.

**Comment by holta** [comment]
>> Can the creation of blank db xklb-metadata.db snapshots be fully automated in future — instead of using a static template/xklb-metadata.db ?
>
> Yes, this is possible

Can this be automated in the near future?  Roughly how many days would this take to get working?

- Certainly let's favor a CI / CD (continuous improvement) approach whenever possible.
- _So that continuous QA is built-in as this evolves._
- Let's not lock into a static schema (if possible!) when xklb's own schema continues to evolve regularly.

**Comment by holta** [comment]
@emg70 & others helping to test:

- Regardless whether this PR #3808 is a temporary "hack" or something more permanent...
- Please consider applying this (PR #3808) to help your testing of the [5+ above-mentioned PRs](https://github.com/iiab/iiab/pull/3808#issuecomment-2374318269), by running...

```
sudo -i
cd /opt/iiab/iiab
git pull https://github.com/deldesir/iiab deldesir-patch-2 --no-rebase --no-edit
./runrole --reinstall calibre-web
```

**Comment by deldesir** [comment]
> @EMG70 & others helping to test:
> 
> * Regardless whether this PR [Add xklb-metadata.db #3808](https://github.com/iiab/iiab/pull/3808) is a temporary "hack" or something more permanent...
> * Please consider applying this (PR [Add xklb-metadata.db #3808](https://github.com/iiab/iiab/pull/3808)) to help your testing of the [5+ above-mentioned PRs](https://github.com/iiab/iiab/pull/3808#issuecomment-2374318269), by running...
> 
> ```
> sudo -i
> cd /opt/iiab/iiab
> git pull https://github.com/deldesir/iiab deldesir-patch-2 --no-rebase --no-edit
> ./runrole --reinstall calibre-web
> ```

`./runrole --reinstall calibre-web` will not touch `/library/calibre-web` AFAIK. If xklb-metadata is already present (from the previous Calibre-Web), a blank xklb-metadata.db will not be copied to it.

**Comment by holta** [comment]
> `./runrole --reinstall calibre-web` will not touch `/library/calibre-web` AFAIK. If xklb-metadata is already present (from the previous Calibre-Web), a blank xklb-metadata.db will not be copied to it.

Good point.  _Are the following instructions better?_

```
sudo -i
cd /opt/iiab/iiab
git pull https://github.com/deldesir/iiab deldesir-patch-2 --no-rebase --no-edit
mv /library/calibre-web/xklb-metadata.db /library/calibre-web/xklb-metadata.db.OLD || true
./runrole --reinstall calibre-web
```

**Comment by deldesir** [comment]
> Can this be automated in the near future? Roughly how many days would this take to get working?
> 
> * Certainly let's favor a CI / CD (continuous improvement) approach whenever possible.
> * _So that continuous QA is built-in as this evolves._
> * Let's not lock into a static schema (if possible!) when xklb's own schema continues to evolve regularly.
 
 Will work on it. Should take less than a couple days.

File: roles/calibre-web/tasks/install.yml [hunk 1]
```diff
@@ -176,6 +176,22 @@
     backup: yes
   when: not appdb.stat.exists
 
+name: Does /library/calibre-web/xklb-metadata.db exist?
```
Comment (holta) [inline roles/calibre-web/tasks/install.yml]: Line 179 contains a bug:

@deldesir please don't announce that a PR was tested if (as in this case) that's not true.

```suggestion
- name: Does /library/calibre-web/xklb-metadata.db exist?
```

Comment (deldesir) [reply roles/calibre-web/tasks/install.yml]: Apologies, I did not catch this as I made the upload and edits directly using Android's Chrome.

**Comment by deldesir** [comment]
Closed in favor of #3827.

**Pull request closed without merging**